qcodes/instrument_drivers/ZI/ZIUHFLI.py

```diff
@@ -2168,6 +2182,23 @@ def getduration():
 
         return value
 
+    @staticmethod
+    def _convert_to_float(frequency):
+        converter = {'hz': 'e0', 'khz': 'e3', 'mhz': 'e6', 'ghz': 'e9',
+                     'thz': 'e12'}
+        value, suffix = frequency.split(' ')
+        return float(''.join([value, converter[suffix.lower()]]))
+
+    def _set_samplingrate_as_float(self, samplingrate_codes, frequency):
+        nearest_frequency = min(samplingrate_codes.keys(), key=lambda k: abs(
+            frequency - ZIUHFLI._convert_to_float(k)))
+
+        self.scope_samplingrate(nearest_frequency)
```

Comment:
smth like `self.scope_samplingrate_float._save_val(ZIUHFLI._convert_to_float(nearest_frequency))` is missing here, right, @QCoDeS/core ?

Reply:
right?

Reply:
@QCoDeS/core ? :)

Reply:
See review below.